migrations/helpers/update_legacy_filtered_positives_helper.py

```diff
@@ -0,0 +1,191 @@
+                " GROUP BY mlwh_sample.description, mlwh_stock_resource.labware_human_barcode, mlwh_sample.phenotype, mlwh_stock_resource.labware_coordinate"  # noqa: E501
+            )
+
+            frame = pd.read_sql(
+                sql,
+                db_connection,
+                params={
+                    "root_sample_ids": tuple(chunk_root_sample_id),
+                    "plate_barcodes": tuple(plate_barcodes),
+                },
+            )
+
+            # drop_duplicates is needed because the same 'root sample id' could
+            # pop up in two different batches, and then it would
+            # retrieve the same rows for that root sample id twice
+            # do reset_index after dropping duplicates to make sure the rows are numbered
+            # in a way that makes sense
+            concat_frame = concat_frame.append(frame).drop_duplicates().reset_index(drop=True)
+
+        return concat_frame
+    except Exception as e:
+        logger.error("Error while connecting to MySQL")
+        logger.exception(e)
+        return None
+    finally:
+        db_connection.close()
+
+
+def v0_version_set(config: ModuleType):
+    """Find if the v0 or v1 version has been set in any of the samples.
+       This would indicate that the legacy migration has already been run.
+
+    Args:
+        samples {List[Dict]} -- List of samples from Mongo
+
+    Returns:
+        Boolean -- v0 version set in samples
+    """
+    with create_mongo_client(config) as client:
+        mongo_db = get_mongo_db(config, client)
+        samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
+
+        v0_samples = list(
+            samples_collection.find({FIELD_FILTERED_POSITIVE_VERSION: FILTERED_POSITIVE_VERSION_0})
+        )  # noqa: E501
+        if len(v0_samples) > 0:
+            return True
+        else:
+            return False
+
+
+def split_mongo_samples_by_version(
+    samples: List[Sample], cp_samples_df_v0: DataFrame, cp_samples_df_v1: DataFrame
+):  # noqa: E501
+    """Split the Mongo samples dataframe based on the v0 cherrypicked samples. Samples
+       which have been v0 cherrypicked need to have the v0 filtered positive rules
+       applied. The remaining samples need the v1 rule applied.
+
+    Args:
+        samples {List[Sample]} -- List of samples from Mongo
+        cp_samples_df_v0: DataFrame -- DataFrame of v0 cherrypicked samples
+        cp_samples_df_v1: DataFrame -- DataFrame of v1 cherrypicked samples
+
+    Returns:
+        samples_by_version {Dict[List[Sample]]} -- Samples split by version
+    """
+    v0_cp_samples = cp_samples_df_v0[[FIELD_ROOT_SAMPLE_ID, FIELD_PLATE_BARCODE]].to_numpy().tolist()  # noqa: E501
+    v1_cp_samples = cp_samples_df_v1[[FIELD_ROOT_SAMPLE_ID, FIELD_PLATE_BARCODE]].to_numpy().tolist()  # noqa: E501
+
+    v0_unmigrated_samples = []
+    v1_unmigrated_samples = []
+    v2_unmigrated_samples = []
```

Comment:
Could you rename these? As far as I understand the workflow can have migrated samples in (e.g. if it fails at a step after Mongo was updated with v0 samples); and this method shouldn't try to know about the migrated state of things - it should just split samples up

Reply:
Renamed 👍